frontend/isoko/src/pages/SearchResults.tsx

```diff
@@ -0,0 +1,541 @@
+      }
+   };
+
+   // build rows of minority tags in the Add Tags modal
+   const renderRows = () => {
+      const fullList = [];
+      const finalArr = [];
+      let columns = [];
+
+      props.minorityTags.forEach((m, i) => {
+         fullList.push(m);
+      });
+
+      // remove tags that are already selected
+      allTags.forEach((t, i) => {
+         if (!props.minorityTags.includes(t)) {
+            fullList.push(t);
+         }
+      });
+
+      fullList.forEach((tag, i) => {
+         if (props.minorityTags.includes(tag)) {
+            columns.push(
+               <ModalTagCol className="col-auto" key={i}>
+                  <StyledButton primary id={tag} onClick={() => addTag(tag)}>
+                     {tag}
+                  </StyledButton>
+               </ModalTagCol>
+            );
+         } else {
+            columns.push(
+               <ModalTagCol className="col-auto" key={i}>
+                  <StyledButton id={tag} onClick={() => addTag(tag)}>
+                     {tag}
+                  </StyledButton>
+               </ModalTagCol>
+            );
+         }
+         // only 4 tags/row
+         if ((i + 1) % 3 === 0) {
+            finalArr.push(<ModalTagRow>{columns}</ModalTagRow>);
+            columns = [];
+         }
+      });
+      finalArr.push(<ModalTagRow>{columns}</ModalTagRow>);
+      return finalArr;
+   };
+
+   return (
+      <SearchContainer>
+         <Row>
+            <Col className="col-sm-3">
+               <Sidebar>
+                  <StyledH2>Sort</StyledH2>
+                  <SidebarContainer>
+                     <StyledDropdownButton title={sortKey}>
+                        <Dropdown.Item as="button">
+                           <div
+                              onClick={(e) => {
+                                 setSortKey(e.target.textContent);
+                                 sortBusinesses('recent');
+                              }}
+                           >
+                              Recently added
+                           </div>
+                        </Dropdown.Item>
+                        <Dropdown.Item as="button">
+                           <div
+                              onClick={(e) => {
+                                 setSortKey(e.target.textContent);
+                                 sortBusinesses('reviews');
+                              }}
+                           >
+                              Most reviews
+                           </div>
+                        </Dropdown.Item>
+                        <Dropdown.Item as="button">
+                           <div
+                              onClick={(e) => {
+                                 setSortKey(e.target.textContent);
+                              }}
+                           >
+                              Highest rated
+                           </div>
+                        </Dropdown.Item>
+                     </StyledDropdownButton>
```

Comment:
Can we pull this into it's own component, just to get this file to be a bit more digestable.